app/lib/tool/backfill/backfill_new_fields.dart

```diff
@@ -24,20 +24,79 @@ Future<void> _backfillPackages() async {
   // during which a new version could be published. To prevent bad updates,
   // the count is not used in such cases, and backfill is re-run every time
   // a race has been detected.
-  var updating = true;
-  while (updating) {
-    updating = false;
-    final query = dbService.query<Package>();
-    await for (final p in query.run()) {
-      final race = await _backfillPackage(p);
-      if (race) updating = true;
+  var wasRace = true;
+  while (wasRace) {
+    wasRace = false;
+
+    await for (final p in dbService.query<Package>().run()) {
+      if (await _packageVersionCount(p)) {
+        wasRace = true;
+      }
+    }
+
+    await for (final p in dbService.query<Package>().run()) {
+      if (await _packagePublishedTimestamps(p)) {
+        wasRace = true;
+      }
     }
   }
```

Comment:
Use package retry throw an exception instead... and if the number of retries is exhausted then we handle that exception print an error and move on.

We should not sprinkle infinite loops around the code. It's better to try a number of times with exponential backoff and otherwise give up.